api/src/main/java/uk/gov/dvsa/mot/persist/jdbc/DvlaVehicleReadSql.java

```diff
@@ -2,29 +2,86 @@
 
 public class DvlaVehicleReadSql {
 
-    static final String selectDvlaVehicleByRegistration = "SELECT `dvla_vehicle`.`id`, `dvla_vehicle`.`dvla_vehicle_id`, " +
-            "`dvla_vehicle`.`registration`, `dvla_vehicle`.`model_code`, `dvla_vehicle`.`make_code`, `dvla_vehicle`.`colour_1_code`, " +
-            "`dvla_vehicle`.`colour_2_code`, `dvla_vehicle`.`manufacture_date`, `dvla_vehicle`.`first_registration_date`, " +
-            "`dvla_vehicle`.`eu_classification`, `dvla_vehicle`.`body_type_code`, `dvla_vehicle`.`last_updated_on` " +
-            "FROM `mot2`.`dvla_vehicle` " +
+    static final String selectDvlaVehicleByRegistration = "" +
+            "SELECT `dvla_vehicle`.`id`,  " +
+            "  `dvla_vehicle`.`dvla_vehicle_id`,  " +
+            "  `dvla_vehicle`.`registration`,  " +
+            "  `dvla_model`.`code` `model_code`, " +
+            "  `dvla_make`.`code` `make_code`, " +
+            "  `model`.`name` `dvsa_model_name`, " +
```

Comment:
FYI - I believe we used a much simpler and more accurate way of obtaining make and model names for MOTH.
Not sure what you end up in your approach if there is no mapping the dvla_model_model_detail_code_map table - probably empty strings.
It removed the need for logic outside of the SELECT for make and model names.

See moth-vehicle-service/src/main/resources/sql/getDvlaVehiclesByRegistration.sql
ie.
  ifnull(vehicle.make_in_full, ifnull(make.name,dvla_make.name)) make_name,
  if (vehicle.make_in_full is null, ifnull(model.name,dvla_model.name), '') model_name,

Reply:
The requirements for the Trade API changes were slightly different. The 'UNKNOWN' value also had to be taken into consideration, and the make_in_full value was required to be returned to MOTR. It seemed easier to add the logic to shared Java code and not complicate the SQL query. However, we can revisit this with a different approach with a new ticket if you think it would better to do more logic in the SQL.